addons/account/models/account_bank_statement_line.py

```diff
@@ -205,10 +216,11 @@ def _compute_running_balance(self):
                 """,
                 [max_index, journal.id] + extra_params,
             )
-            for st_line_id, amount, is_anchor, balance_start in self._cr.fetchall():
+            for st_line_id, amount, is_anchor, balance_start, state in self._cr.fetchall():
                 if is_anchor:
                     current_running_balance = balance_start
-                current_running_balance += amount
+                if state == 'posted':
+                    current_running_balance += amount
```

Comment:
Something like that is more logical to me:
```py
for st_line_id, amount, is_anchor, balance_start, state in self._cr.fetchall():
     if state != 'posted':
        continue
    ...
```
What do you think?

Reply:
What if a user cancels a line which is the anchor line of a statement?